front/app/containers/Admin/reporting/components/ReportBuilder/Toolbox/index.tsx

```diff
@@ -79,6 +114,12 @@ const ReportBuilderToolbox = ({ reportId }: ReportBuilderToolboxProps) => {
     });
   };
 
+  // If this report is not in a phase context (i.e. projectId is undefined),
+  // AND the user is moderator (i.e. projects is defined),
+  // we use the first project in the list of projects as the default project.
+  const selectedProjectId =
+    projectId ?? isModerator ? projects?.data[0]?.id : undefined;
```

Comment:
And what if `projectId` is present? I guess we should use it as `selectedProjectId`, right? But it seems now we use `projects?.data[0]?.id` in this case
```js
projectId = '123'
isModerator = true
projectId ?? isModerator ? 1 : 2 // => 1
```
```suggestion
    projectId ?? (isModerator ? projects?.data[0]?.id : undefined);
```

Reply:
Damned order of operations! Great catch :+1: